python-threatexchange/threatexchange/exchanges/impl/ncmec_api.py

```diff
@@ -303,12 +331,23 @@ def log(event: str) -> None:
                         # Our entry estimatation (based on the cursor parameters)
                         # occasionally seem to over-estimate
                         log(f"est {entry.estimated_entries_in_range} entries")
-                elif i % 100 == 0:
-                    # If we get down to one second, we can potentially be
-                    # fetching an arbitrary large amount of data in one go,
-                    # so log something occasionally
-                    log(f"large fetch ({i}), up to {len(updates)}")
+
                 updates.extend(entry.updates)
+
+                if i > 100 and len(updates) > 0:
+                    # On large fetches, yield a checkpoint on each iteration
+                    log(
+                        f"large fetch ({i}) with {len(updates)} updates. yielding checkpoint"
+                    )
+                    yield state.FetchDelta(
+                        {f"{entry.member_id}-{entry.id}": entry for entry in updates},
+                        NCMECCheckpoint(
+                            get_entries_max_ts=current_start,
+                            _paging_url=entry.next,
+                        ),
+                    )
+                    updates = []
```

Comment:
Hmm, What is your intent here? 

It will buffer updates without emitting them until 100 fetches, and then after it will emit every update? 
```suggestion

                if i % 100 == 5:
                    # On large fetches, log notice every once in a while
                    log(
                        f"large fetch ({i}) with {len(updates)} updates"
                    )
                yield state.FetchDelta(
                    {f"{entry.member_id}-{entry.id}": entry for entry in updates},
                    NCMECCheckpoint(
                        get_entries_max_ts=current_start,
                        _paging_url=entry.next,
                    ),
                )
                updates = []
```

Reply:
how do we know when we have a "large update"? before that was i > 100, but if we're fine considering the 5th iteration to be large, then this makes sense! 

There doesn't seem to be a downside to yielding, so it probably doesn't matter, right?

Reply:
in the proposed code, there's no need to

`updates.extend(entry.updates)`, right? since we're always resetting it back to an empty array?